lldb-eval/parser.cc

```diff
@@ -1780,7 +1816,102 @@ ExprResult Parser::ParseIntegerLiteral(clang::NumericLiteralParser& literal,
       CreateValueFromAPInt(target_, llvm::APSInt(raw_value, is_unsigned),
                            target_.GetBasicType(type));
 
-  return std::make_unique<LiteralNode>(std::move(value));
+  return std::make_unique<LiteralNode>(token.getLocation(), std::move(value));
+}
+
+// Parse a builtin_func.
+//
+//  builtin_func:
+//    builtin_func_name "(" {builtin_func_argument_list} ")"
+//
+//  builtin_func_name:
+//    "__log2"
+//
+//  builtin_func_argument_list:
+//    builtin_func_argument
+//    builtin_func_argument_list "," builtin_func_argument_list
+//
+//  builtin_func_argument:
+//    expression
+//
+ExprResult Parser::ParseBuiltinFunction(
+    std::unique_ptr<BuiltinFunctionDef> func_def) {
+  // Check for builtin debugger intrinsic function.
+  clang::SourceLocation loc = token_.getLocation();
+  ConsumeToken();
+
+  Expect(clang::tok::l_paren);
+  ConsumeToken();
+
+  std::vector<ExprResult> arguments;
+
+  do {
+    // Eat the comma if this is not the first iteration.
+    if (arguments.size() > 0) {
+      ConsumeToken();
+    }
+
+    // Parse a builtin_func_argument. If failed to parse, bail out early and
+    // don't try parsing the rest of the arguments.
+    auto argument = ParseExpression();
+    if (argument->is_error()) {
+      return std::make_unique<ErrorNode>();
+    }
+
+    arguments.push_back(std::move(argument));
+
+  } while (token_.is(clang::tok::comma));
+
+  Expect(clang::tok::r_paren);
+  ConsumeToken();
+
+  // Check we have the correct number of arguments.
+  if (arguments.size() != func_def->arguments_.size()) {
```

Comment:
We could also bail out a bit earlier in the case there are more arguments than expected with additional check in the do-while loop. Is that going to make a difference?

Reply:
We need to parse all the arguments so that we can have an error message like "requires N arguments, got K arguments".

Reply:
Makes sense, thanks for clarification.